tests/test_stack.py

```diff
@@ -191,10 +191,37 @@ def test_stack_tree(hlwm):
         - Client <windowid> "bash"
       - Frame Layer
         - Window <windowid>
+    - Desktop Windows
 '''
     assert strip_winids(stack.stdout) == expected_stack
 
 
+def test_stack_tree_desktop_windows(hlwm, x11):
+
+    _, winid = x11.create_client(wm_class=('TestDesktop', 'TestDeskClass'),
+                                 window_type='_NET_WM_WINDOW_TYPE_DESKTOP')
+
+    # Simplified tree style:
+    hlwm.call('set tree_style "     - -"')
+    framewindows = [x11.winid_str(w) for w in x11.get_hlwm_frames()]
+    expected_stack = f'''\
+  -
+    - Monitor 0 with tag "default"
+      - Focus-Layer
+      - Fullscreen-Layer
+      - Floating-Layer
+      - Tiling-Layer
+      - Frame Layer
+        - Window {framewindows[0]}
+    - Desktop Windows
+      - {winid} TestDesktop TestDeskClass
+'''
+    stack = hlwm.call('stack')
+    print("expected:")
+    print(expected_stack)
```

Comment:
Do you want to keep those print statements?
The assert statement should print the diff between expected and actual anyway, right?

Reply:
I forgot to remove the print-statement, but I had added it because the diff wasn't helpful at all. During writing, I forgot to apply `x11.winid_str(w)` in line 206 and so the content of 'Frame Layer' was wrong. The output then was:
```
>       assert stack.stdout == expected_stack
E       assert '  -\n    - M...stDeskClass\n' == '  -\n    - M...stDeskClass\n'
E             -
E               - Monitor 0 with tag "default"
E                 - Focus-Layer
E                 - Fullscreen-Layer
E                 - Floating-Layer
E                 - Tiling-Layer
E                 - Frame Layer...
E         
E         ...Full output truncated (6 lines hidden), use '-vv' to show

tests/test_stack.py:220: AssertionError
```
Do I have to activate something to obtain a proper diff?

Anyway, I have removed the print-statement, thanks for spotting :)

Reply:
The full diff only comes when you run pytest with -vv.
What I usually do, when a test fails I run it explicitly with  ```pytest -vv test_foo.py:test_foo```